services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy

```diff
@@ -221,4 +225,361 @@ class MflStatSpec extends HealthCheckSpecification {
+        assert traffExam.waitExam(exam).hasTraffic()
+
+        then: "Stats is not empty for main path cookies"
+        def metric = metricPrefix + "flow.raw.bytes"
+        def tags = [switchid: switchPair.src.dpId.toOtsdFormat(), flowid: flow.id]
+        def waitInterval = 10
+        def mainPathStat
+        Wrappers.wait(statsRouterInterval, waitInterval) {
+            mainPathStat = otsdb.query(startTime, metric, tags).dps
+            assert mainPathStat.size() >= 1
+        }
+        def flowInfo = database.getFlow(flow.id)
+        def mainForwardCookie = flowInfo.forwardPath.cookie.value
+        def mainReverseCookie = flowInfo.reversePath.cookie.value
+        def mainForwardCookieStat = otsdb.query(startTime, metric, tags + [cookie: mainForwardCookie]).dps
+        def mainReverseCookieStat = otsdb.query(startTime, metric, tags + [cookie: mainReverseCookie]).dps
+        [mainForwardCookieStat, mainReverseCookieStat].each { stats ->
+            stats.values().each { assert it != 0 }
+        }
+
+        and: "Stats is empty for protected path egress cookie"
+        def protectedReverseCookie = flowInfo.protectedReversePath.cookie.value
+        def protectedReverseCookieStat = otsdb.query(startTime, metric, tags + [cookie: protectedReverseCookie]).dps
+        protectedReverseCookieStat.values().each { assert it == 0 }
+
+        when: "Swap main and protected path"
+        northbound.swapFlowPath(flow.id)
+        Wrappers.wait(PROTECTED_PATH_INSTALLATION_TIME) {
+            assert northbound.getFlowStatus(flow.id).status == FlowState.UP
+            def newFlowPathInfo = northbound.getFlowPath(flow.id)
+            assert pathHelper.convert(newFlowPathInfo) == currentProtectedPath
+            assert pathHelper.convert(newFlowPathInfo.protectedPath) == currentPath
+        }
+
+        and: "Generate traffic on the flow"
+        exam.setResources(traffExam.startExam(exam, true))
+        assert traffExam.waitExam(exam).hasTraffic()
+
+        then: "Stats is not empty for previous protected egress cookie"
+        Wrappers.wait(statsRouterInterval, waitInterval) {
+            def protectedPathStat = otsdb.query(startTime, metric, tags).dps
+            assert protectedPathStat.size() > mainPathStat.size()
+            def newProtectedReverseCookieStat = otsdb.query(startTime, metric, tags + [cookie: protectedReverseCookie]).dps
+            assert !newProtectedReverseCookieStat.values().findAll { it != 0 }.empty
+        }
+
+        and: "Stats is not changed for previous main path cookies"
+        def newMainForwardCookieStat = otsdb.query(startTime, metric, tags + [cookie: mainForwardCookie]).dps
+        def newMainReverseCookieStat = otsdb.query(startTime, metric, tags + [cookie: mainReverseCookie]).dps
+        newMainForwardCookieStat.size() == mainForwardCookieStat.size()
+        //sometimes newMainReverseCookieStat.size() = mainReverseCookieStat.size() + 1
```

Comment:
Check point values instead of points amount? New point value should equal to old point value.

Reply:
decided to change verification
now we are going to check the following:
```
  and: "Stats is still being collected for previous egress cookie of main path"
 and: "System doesn't collect stats anymore for previous ingress cookie of main path"
```

Reply:
decided to change verification
now we are going to check the following:
```
  and: "Stats is still being collected for previous egress cookie of main path"
 and: "System doesn't collect stats anymore for previous ingress cookie of main path"
```